### PR TITLE
[WIP] New packages: proton-bridge-1.2.7 and dependency

### PR DESCRIPTION
**Pull request by cinerea0**
I'm trying to package [proton-bridge](https://github.com/ProtonMail/proton-bridge), but I'm running into some issues with the build process. Here's the relevant output from xbps-src:
```
make: qtdeploy: No such file or directory
make: *** [Makefile:66: cmd/Desktop-Bridge/deploy/linux/proton-bridge-1.2.7-live.1] Error 127
=> ERROR: proton-bridge-1.2.7_1: do_build: '${make_cmd} CC="$CC" CXX="$CXX" LD="$LD" AR="$AR" RANLIB="$RANLIB" CPP="$CPP" AS="$AS" OBJDUMP="$OBJDUMP" CFLAGS="$CFLAGS" LDFLAGS="$LDFLAGS" ${makejobs} ${make_build_args} ${make_build_target}' exited with 2
=> ERROR:   in do_build() at common/build-style/gnu-makefile.sh:9
```
I know that xlint throws up errors on this template as well, I thought I should fix those after solving the build errors. Any other suggestions for improvement are welcome as well!

As for `qtdeploy`, the best I can figure is that it's a tool provided by [this repository](https://github.com/therecipe/qt). Unfortunately, the repo has no releases, so the bindings and tools it provides can't be packaged. ericonr on IRC said that it might be possible to build without `qtdeploy`, but I haven't understood the Makefile well enough yet to figure out how to do that.

### CODE REVIEW

**Comment by ericonr** [comment]
I corrected myself later, it seems like qtdeploy is indeed necessary for it, which is a damn shame. The Makefile calls it directly. 

`qtdeploy` is here: https://github.com/therecipe/qt/blob/master/cmd/qtdeploy . No releases so far.

**Comment by cinerea0** [comment]
That leaves things in a bit of a bind. I've opened an issue in that repository to request a release, but I have no idea if it will be answered.

**Comment by ProjectMoon** [comment]
I think the main reason there's no release of qtdeploy is because it's meant to be used for building go applications as a library/module thingy. Then you upload your super cool static binary somewhere for people to download.

But seeing as how the Go QT bindings have been around for quite awhile and don't seem to show any signs of making a proper release as such, wouldn't this be one of the good reasons to package qtdeploy despite it not having a proper version as such?

To be honest I find it weird that they don't at least tag stuff in that repository.

**Comment by cinerea0** [comment]
@ProjectMoon To clarify, are you saying that this could be a situation where it would be appropriate to package the bindings or just qtdeploy in spite of the "must be tied to a release" rule?

**Comment by ProjectMoon** [comment]
Yes. Or use qtdeploy locked to a specific tag in the build for this package.

**Comment by ericonr** [comment]
If we could use the same commit as the ones tagged in the docker images, that would be great. Then document this in the template.

**Comment by cinerea0** [comment]
@ericonr I'll get to work packaging those go bindings, then. I'm not familiar with standard procedure in cases like this: Should I close this PR, open a new one for the bindings once I have it finished, and then open a PR for proton-bridge once the bindings have been accepted, or should I add the template for the bindings to this pull request?

**Comment by ericonr** [comment]
@cinerea0 since it's a tool to use in this build, I'd put it in this same PR.

**Comment by cinerea0** [comment]
Just wanted to give an update, this is currently blocking on ProtonMail/proton-bridge#52 because proton-bridge fails to build either manually or with xbps-src. I have managed to package qt5-go successfully, though.

**Comment by ProjectMoon** [comment]
I was able to build the bridge some time ago. But I had the go qt bindings installed via Go.

**Comment by Chocimier** [comment]
> pkg-config: exec: "pkg-config": executable file not found in $PATH

Looks like missing pkg-config.

**Comment by cinerea0** [comment]
I added pkg-config in "hostmakedepends", and I also added "libsecret" because that error came up afterwards. However, that resulted in the following error:
```
# pkg-config --cflags  -- libsecret-1
Package libsecret-1 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libsecret-1.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libsecret-1' found
pkg-config: exit status 1
```
Does this mean I have to do some patching to rename "libsecret" to "libsecret-1"? Also, I've added the most recent version of the templates for review.

**Comment by Chocimier** [comment]
libsecret-1.pc is part of libsecret-devel.

pkg-config, git and other packages installed for executables should go into hostmakedepends. gcc, make are always available during build as part of base-chroot.

Compiler output are warnings, not error.

**Comment by cinerea0** [comment]
Thank you for the tip about libsecret-devel! That got rid of that error and all the compiler warnings, but I'm now encountering this new one:
`make: *** No rule to make target 'install'.  Stop.`
As a side note, I have a manual post_install function that just does `vlicense LICENSE`. Is that necessary, or does xbps-src take care of that automatically if the license is just called "LICENSE"?

**Comment by cinerea0** [comment]
Upon updating the templates, I'm getting two new errors:
`ERROR: proton-bridge-1.2.7_1: failed to install 'go>=1.13' dependency! (error 34)`
`ERROR: qt5-go-0.1_1: SHA256 hash is not valid: Numerical result out of range`
Additionally, how should icons be handled in the `do_install` function?

File: srcpkgs/proton-bridge/template [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Template file for 'proton-bridge'
+pkgname=proton-bridge
+version=1.2.7
+revision=1
+archs="*"
```
Comment (abenson) [inline srcpkgs/proton-bridge/template:5]: Remove this

**Comment by abenson** [comment]
Each template should have its own commit, but both should be in this PR

File: srcpkgs/qt5-go/template [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'qt5-go'
+pkgname=qt5-go
+version=0.1
+revision=1
+archs="*"
+wrksrc=qt-b56356132618ecc7c0029f4262d4ac986afe1477
```
Comment (abenson) [inline srcpkgs/qt5-go/template:6]: the hash should be a variable, like `_githash` and then when needed use `${_githash}` instead to avoid repetition

File: srcpkgs/qt5-go/template [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'qt5-go'
+pkgname=qt5-go
+version=0.1
+revision=1
+archs="*"
```
Comment (abenson) [inline srcpkgs/qt5-go/template:5]: Same as above

File: srcpkgs/qt5-go/template [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'qt5-go'
+pkgname=qt5-go
+version=0.1
```
Comment (abenson) [inline srcpkgs/qt5-go/template:3]: Use something like 0.0.2020.06.04

File: srcpkgs/qt5-go/template [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+wrksrc=qt-b56356132618ecc7c0029f4262d4ac986afe1477
+#create_wrksrc=yes
+build_style=go
+go_import_path=github.com/therecipe/qt/cmd/...
+#configure_args=""
+#make_build_args=""
+#make_install_args=""
+#conf_files=""
+#make_dirs="/var/log/dir 0755 root root"
+hostmakedepends="git"
```
Comment (abenson) [inline srcpkgs/qt5-go/template:15]: This doesn't seem to be needed

File: srcpkgs/proton-bridge/template [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+wrksrc="proton-bridge-${version}-live.1"
+#create_wrksrc=yes
+build_style=gnu-makefile
+#configure_args=""
+#make_build_args=""
+#make_install_args=""
+#conf_files=""
+#make_dirs="/var/log/dir 0755 root root"
+hostmakedepends=""
+makedepends="binutils git go>=1.13 libsecret-devel pkg-config qt5-go tar which"
```
Comment (abenson) [inline srcpkgs/proton-bridge/template:15]: `binutils` `go` `pkg-config` `qt5-go` `tar` `which` should all go in `hostmakedepends`

File: srcpkgs/proton-bridge/template (hunk 1 above)
Comment (cinerea0) [reply srcpkgs/proton-bridge/template:5]: All four lines or just the archs line?

File: srcpkgs/proton-bridge/template [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+short_desc="ProtonMail Bridge for e-mail clients"
+maintainer="cinerea0 <cinerea0@protonmail.com>"
+license="GPL-3.0-or-later"
+homepage="https://protonmail.com/bridge"
+distfiles="https://www.github.com/ProtonMail/proton-bridge/archive/v${version}-live.1.tar.gz"
+checksum=5a4c19b769cd72097507de6840b79e56d37fa6a54cf7269d4fec28c05016ca45
+
+do_install() {
+    vbin ./cmd/Desktop-Bridge/deploy/linux/proton-bridge-1.2.7-live.1 proton-bridge
+    vdoc ./doc/*
```
Comment (abenson) [inline srcpkgs/proton-bridge/template:26]: Quote globs (`vbin "./doc/*"`)

**Comment by abenson** [comment]
I've gotten it to this point

    => proton-bridge-1.2.7_1: running post-install hook: 06-strip-and-debug-pkgs ...
       Stripped executable: /usr/bin/proton-bridge
    => ERROR: proton-bridge-1.2.7_1: non-PIE executable found in PIE build: /usr/bin/proton-bridge

Which means it's not respecting the flags in the environment.

I tried to build it using the go build_helper but that seems to have its own issues

File: srcpkgs/proton-bridge/template (hunk 1 above)
Comment (cinerea0) [reply srcpkgs/proton-bridge/template:5]: Never mind, you meant the archs line.

**Comment by cinerea0** [comment]
@abenson I've made the changes you suggested with the exception of the `vdoc` one because I haven't been able to test it, but I'm still some strange output that results in an error:
```
index: skipping `qt5-go-0.0.2020.06.04_1' (x86_64), already registered.
index: 1 packages registered.
=> qt5-go-0.0.2020.06.04_1: removing autodeps, please wait...
=> qt5-go-0.0.2020.06.04_1: cleaning build directory...
=> qt5-go: removing files from destdir...
=> proton-bridge-1.2.7_1: installing host dependencies: go>=1.13 binutils-2.34_1 pkg-config-0.29.2_2 tar-1.32_3 which-2.21_3 qt5-go-0.0.2020.06.04_1 ...
=> proton-bridge-1.2.7_1: removing autodeps, please wait...
=> ERROR: proton-bridge-1.2.7_1: failed to install 'go>=1.13' dependency! (error 2)
Package 'qt5-go-0.0.2020.06.04_1' not found in repository pool.
Package `binutils-2.34_1' already installed.
=> ERROR: Please see above for the real error, exiting...
```
It looks like it's building qt5-go everytime I try to build proton-bridge. I think it might be an xbps-src issue? Regardless, I'll update my open issue in the proton-bridge repository to reflect your updated error status.

**Comment by ericonr** [comment]
@abenson the go build style sets nostrip and nopie. It is kind of necessary for now, because Go binaries shouldn't be stripped and PIE isn't enabled by default in upstream (see #20854).

@cinerea0 abenson managed to build it, that error is from xbps-src itself, because the template should have the `nostrip` and `nopie` variables set to `yes`.

**Comment by abenson** [comment]
> the go build style sets nostrip and nopie

I thought that might be the case but in my quick search I didn't see it

**Comment by cinerea0** [comment]
The package builds completely now, but upon running it I get this error:
```
INFO[0000] Run app                                       appLong="Protonmail Bridge" appShort=bridge args="[proton-bridge]" build="2020-06-11T04:06:22+0000" pkg=main revision= runtime=linux version=1.2.7-git
proton-bridge: symbol lookup error: proton-bridge: undefined symbol: _ZdlPvm, version Qt_5
```

**Comment by ericonr** [comment]
That sounds ugly, it should have resolved stuff at build time. Do you have `qt5` installed? It should likely be in depends.

**Comment by cinerea0** [comment]
@ericonr Yes, I have qt5 installed. I have quite a few qt libraries installed because I use KDE.

File: srcpkgs/proton-bridge/template [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'proton-bridge'
+pkgname=proton-bridge
+version=1.2.7
+revision=1
+wrksrc="proton-bridge-${version}-live.1"
+build_style=gnu-makefile
+#make_dirs="/var/log/dir 0755 root root"
+hostmakedepends="binutils git go pkg-config qt5-go tar which"
```
Comment (pullmoll) [inline srcpkgs/proton-bridge/template:8]: Remove `binutils` - it is part of `base-chroot` and always available.

**Comment by ericonr** [comment]
@cinerea0 I don't know how the build system works, but I'd try to put stuff like qt5-devel in makedepends, and see if it works. If it doesn't, then install all the qt5 packages you can think of or try to search for that error online (quote the missing symbol).

**Comment by cinerea0** [comment]
@ericonr Adding qt5-devel to makedepends did not fix it. After searching for that symbol and doing some investigating, I think I may have discovered the problem, but I'm not sure it's one that can be fixed. The [closest related link](https://www.reddit.com/r/linux_gaming/comments/ginkmo/ubuntu_2004_tibia_resulting_in_libqxcbso/) was from a Linux gaming forum, and the comments there described the problem as potentially arising from a library mismatch. When proton-bridge builds, it pulls in an environment (I only noticed this because it takes a while during the build process). When I investigated the release tarball, I found vendor dependencies from therecipe (same guy who made the go bindings) for various qt libraries, all of which are at version 5.13.0. The Void versions of those libraries are all at version 5.14.2. Additionally, it might not be a mismatch in the exact libraries, but in the version of gcc that was used to compile them. 

Basically, what I'm getting at is that there needs to be a way to use system dependencies instead of that environment it pulls in.

**Comment by cinerea0** [comment]
I found this weird script in the environment directory, it seems to be the one drawing in the 5.13.0 libraries:
```bash
#!/bin/bash

set -ev

QT_ROOT=/opt
QT_VERSION=5.13.0

rm -rf ./${QT_VERSION}
rm -rf ./Licenses


rsync -avz $QT_ROOT/Qt/${QT_VERSION}/gcc_64 ./${QT_VERSION}/
rsync -avz $QT_ROOT/Qt/Licenses .


rm -rf ./${QT_VERSION}/gcc_64/{doc,phrasebooks}
rm -rf ./${QT_VERSION}/gcc_64/lib/{cmake,pkgconfig,libQt5Bootstrap.a}


for v in *.jsc *.log *.pro *.pro.user *.qmake.stash *.qmlc .DS_Store *_debug* *.la *.prl; do
	find . -name ${v} -exec rm -rf {} \;
done

mkdir -p ./${QT_VERSION}/gcc_64/_bin
for v in moc qmake qmlcachegen qmlimportscanner qt.conf rcc uic; do
	mv ./${QT_VERSION}/gcc_64/bin/${v} ./${QT_VERSION}/gcc_64/_bin/
done
rm -rf ./${QT_VERSION}/gcc_64/bin && mv ./${QT_VERSION}/gcc_64/_bin ./${QT_VERSION}/gcc_64/bin

find ./${QT_VERSION}/gcc_64/bin -type f ! -name "qt.conf" -exec strip -s {} \;
find ./${QT_VERSION}/gcc_64/lib -type f ! -name "*.a" -name "lib*" -exec strip -s {} \;
find ./${QT_VERSION}/gcc_64/lib -type f -name "lib*.a" -exec strip -S {} \;
find ./${QT_VERSION}/gcc_64/plugins -type f -name "lib*" -exec strip -s {} \;
find ./${QT_VERSION}/gcc_64/qml -type f -name "lib*" -exec strip -s {} \;

mv $QT_ROOT/Qt $QT_ROOT/Qt_orig

go run ./patch.go

gzip -n ./${QT_VERSION}/gcc_64/lib/libQt5WebEngineCore.so.${QT_VERSION}

du -sh ./5*

#$(go env GOPATH)/bin/qtsetup
```
Could it be possible to create a custom version of this script, put it in the package, and patch the Makefile to invoke the script somewhere around here:
```make
vendor-cache/${THERECIPE_ENV}:
	git clone https://${THERECIPE_ENV}.git vendor-cache/${THERECIPE_ENV}
```

**Comment by ProjectMoon** [comment]
When I built this thing locally, I definitely did it against a local Qt installation rather than the system one, I think.

**Comment by cinerea0** [comment]
@ProjectMoon To clarify, do you mean that the library/gcc mismatch probably isn't the problem?

**Comment by ProjectMoon** [comment]
I'm not sure. All I remember is that I built against a qt installation from qt.io specifically because the ones on void were newer. I think I also built against system libs. But the whole thing was outside of an XBPS chroot

**Comment by ericonr** [comment]
@cinerea0 hells that thing is ugly. Ok, the repo being pulled in is full of pre compiled stuff and some Go files. I honestly have no idea how to deal with this. 
https://github.com/therecipe/env_linux_amd64_513

Take a look at the "Official version" and "Your system package managers version" (likely this last one) in https://github.com/therecipe/qt/wiki/Installation-on-Linux . That might be your fix.

**Comment by cinerea0** [comment]
@ericonr I think that would mainly affect qt5-go, but I'm willing to give it a shot. Two problems:
1. I have no idea which Void packages contain the html/doc .index files that are mentioned in the package manager method.
2. The setup requires using a binary that is included in the package. Is that even remotely doable with xbps-src?

**Comment by ericonr** [comment]
@cinerea0 I think that should fix qt5-go enough that it will work properly when building proton-bridge. Regarding those files, check the `qt5-devel` and `qt5-doc` packages. Possibly `qt5-quickcontrols2-devel` or `qt5-quickcontrols` and `qt5-declarative`. Could need `pkg-config` in hostmakedepends.

Running a binary inside xbps-src should be possible, just tends to make stuff nocross.  There's a chance the qtsetup step is for building proton-bridge instead, but I'm not entirely sure.

**Comment by cinerea0** [comment]
I've attempted to translate the qt5-go installation instructions into an xbps template as best I can, but there's a couple hangups. Focusing on this part of the template:
```bash
post_install() {
	$(go env GOPATH)/bin/qtsetup
}
```
There are two possible errors depending on if that block exists or not. With it, the error looks like this: `/void-packages/srcpkgs/qt5-go/template: line 24: /builddir/qt-b56356132618ecc7c0029f4262d4ac986afe1477/_build-qt5-go-xbps/bin/qtsetup: No such file or directory`. Without it, the error looks like this: `ERROR: qt5-go-0.0.2020.06.04_1: PKGDESTDIR is empty and build_style != meta`.

**Comment by cinerea0** [comment]
Alright, I think I've improved the qt5-go template as much as I can. It turns out invoking `qtsetup` was actually ruining the package because it was turning it into a local install as opposed to a system-wide install. However, the error with the undefined symbol still exists. At this point, I think the best way forward is for me to focus on patching the build scripts in such a way that the project is tricked into linking against system libraries.

**Comment by fosslinux** [comment]
Yes, that is precisely the best option here, to patch the build scripts :)

**Comment by cinerea0** [comment]
Looking through the Makefile, there's a couple things I need to get straight before attempting to patch. For one, I have no idea what these lines are doing:
```makefile
.PHONY: prepare-vendor update-vendor
THERECIPE_QTVER:=$(shell grep "github.com/therecipe/qt " go.mod | sed -r 's;.* v[0-9\.]+-[0-9]+-([a-f0-9]*).*;\1;')
```
Next, is it possible for me to prevent this block from being executed simply by removing all references to it:
```makefile
prepare-vendor:
	go install -v -tags=no_env github.com/therecipe/qt/cmd/...
	go mod vendor
```
Finally, these sections are the ones that deal with pulling in the environment libraries:
```makefile
# vendor folder will be deleted by gomod hence we cache the big repo
# therecipe/env in order to download it only once
vendor-cache/${THERECIPE_ENV}:
	git clone https://${THERECIPE_ENV}.git vendor-cache/${THERECIPE_ENV}

LINKCMD:=ln -sf ${CURDIR}/vendor-cache/${THERECIPE_ENV} vendor/${THERECIPE_ENV}

# update-vendor is PHONY because we need to make sure that we always have updated vendor
update-vendor: vendor-cache/${THERECIPE_ENV} prepare-vendor
	${LINKCMD}
```
My plan is to allow these blocks to execute, but afterwards to replace the contents of the environment directory with libraries copied from wherever xbps-src puts them during the build process. Does that make sense?

**Comment by cinerea0** [comment]
I've made my first attempt at creating the patch files. During the patching process, I'm encountering the following error:
```
=> proton-bridge-1.2.7_1: patching: init.sh.patch.
=> ERROR: proton-bridge-1.2.7_1: do-patch_00-patches: 'patch -sl ${_args} -i ${_patch} 2> /dev/null' exited with 2
=> ERROR:   in _process_patch() at common/hooks/do-patch/00-patches.sh:34
=> ERROR:   in hook() at common/hooks/do-patch/00-patches.sh:51
=> ERROR:   in run_func() at common/xbps-src/shutils/common.sh:21
=> ERROR:   in run_pkg_hooks() at common/xbps-src/shutils/common.sh:232
=> ERROR:   in run_step() at common/xbps-src/shutils/common.sh:71
=> ERROR:   in main() at common/xbps-src/libexec/xbps-src-dopatch.sh:33
```
I think I've constructed my patch file for init.sh improperly, but I'm not quite sure how.

File: srcpkgs/proton-bridge/patches/Makefile.patch [hunk 9]
```diff
@@ -0,0 +1,159 @@
+--- Makefile	2020-05-18 08:02:29.000000000 -0400
++++ "Makefile copy"	2020-06-19 21:28:24.937942084 -0400
```
Comment (ericonr) [inline srcpkgs/proton-bridge/patches/Makefile.patch:2]: Filenames need to be the same. You should leave a comment on the patch to explain what it's about, btw.

Comment (Chocimier) [reply srcpkgs/proton-bridge/patches/Makefile.patch:2]: > Filenames need to be the same.

Nope, you often see `Makefile` vs ` Makefile~` or so and it works.

File: srcpkgs/proton-bridge/patches/init.sh.patch [hunk 10]
```diff
@@ -0,0 +1,4 @@
+5c5
```
Comment (Chocimier) [inline srcpkgs/proton-bridge/patches/init.sh.patch:1]: Not specified which file to patch. did you created this file with `diff oldfile newfile`?

Comment (Chocimier) [reply srcpkgs/proton-bridge/patches/init.sh.patch:1]: try `-u` then, from wrksrc level directory

**Comment by cinerea0** [comment]
Updated the patch files with suggestions, now I'm getting a new error:
```
=> proton-bridge-1.2.7_1: patching: init.sh.patch.
The text leading up to this was:
--------------------------
|--- vendor-cache/github.com/therecipe/env_linux_amd64_513/init.sh	2020-06-05 22:39:59.740997705 -0400
|+++ "init copy.sh"	2020-06-19 21:32:02.000000000 -0400
--------------------------
File to patch:
Skip this patch? [y]
1 out of 1 hunk ignored
=> ERROR: proton-bridge-1.2.7_1: do-patch_00-patches: 'patch -sl ${_args} -i ${_patch} 2> /dev/null' exited with 1
```

**Comment by cinerea0** [comment]
Okay, I finally realized what was going on. It turns out I'm trying to patch a file that doesn't exist until after the build process starts. Is there any way to force a patch during the middle of the build process?

**Comment by ericonr** [comment]
Can you manually drive the build process to stop in the middle of that step? Otherwise you'll have to do something like patch the makefile to run your command the moment it needs to be run.

**Comment by fosslinux** [comment]
Generally, the best idea would be to find out where the file is coming from in the build process, and patch the input. For example, you would patch `Makefile.in`, not `Makefile` in a configure-style tarball.

**Comment by fosslinux** [comment]
Oooh, I see, it's from a pulled remote. I really hate this, Rust and Go both do it and it makes a pain.

Here's how to approach this:

1. Make `env_linux_amd64_513` tarball a distfile. Download and extract it.
2. Apply the patch to the extracted tarball.
3. Patch the Makefile for `THERECIPE_ENV` on line 83 to point to the extracted tarball.

Then go will use your patch.

**Comment by cinerea0** [comment]
There still seems to be problems with xbps-src finding the `init.sh` file. Is my directory structure wrong?

**Comment by cinerea0** [comment]
Good news: I figured out the patching. Turns out I just needed to make more judicious use of the wrksrc variable and how it changes.
Bad news: The init.sh script now needs more patching. Turns out the library structure that exists in the xbps-src chroot is a lot different from whatever the dev's environment is.

**Comment by cinerea0** [comment]
I'm deciding to close this (for now) for three reasons:

1. I realized that the function of this package can be easily replicated by setting up [hydroxide](https://github.com/emersion/hydroxide) as a per-user service.
2. We already have [a restricted version of this package that provides the same program](https://github.com/void-linux/void-packages/blob/master/srcpkgs/protonmail-bridge/template).
3. This thing's build process is still a nightmare, and judging by the new releases that isn't changing anytime soon.

I'll still be keeping tabs on new releases, and if the build process improves I'll make another attempt at packaging.

**Comment by ProjectMoon** [comment]
It should be noted that hydroxide isn't a 100% solution. It can be a bit flaky and doesn't support attachments (at time of writing).
